src/main/java/use_cases/display_recipe_detail/DisplayRecipeDetailSearchResultView.java

```diff
@@ -103,33 +102,48 @@ public JPopupMenu showAddToMenu(Recipe recipe) {
 
 
     private void createNewGroceryListAndAdd(Recipe recipe) {
-        String newListName = JOptionPane.showInputDialog((JFrame) SwingUtilities.getWindowAncestor(this), "Enter name for new grocery list:");
-        //need to handle when this is empty
-        if (newListName != null && !newListName.trim().isEmpty()) {
+        String newListName = JOptionPane.showInputDialog((JFrame) SwingUtilities.getWindowAncestor(this),
+                "Enter name for new grocery list:");
+        if (newListName == null) {
```

Comment:
whats this?

Reply:
(im talking about the if on 107 and it just leading to a return on 108

Reply:
this gets rid of the error message : Cannot invoke "String.trim()" because "newListName" is null

which happens when user doesn't put in anything for the name 

but you're right, i'm not sure why i made that a separate case lol. i will collapse it with the else if right below.